frontend/src/views/ProjectView.tsx

```diff
@@ -1,22 +1,72 @@
-import ProjectList from "../components/ProjectList";
-import "../styles/ProjectView.css";
-import "../styles/ProjectList.css";
-import CreateProjectDialog from "../components/CreateProjectDialog";
-import { Token } from "../contexts/Auth.tsx";
+import { useState } from 'react';
+import '../styles/CreateProjectDialog.css';
+import { useParams } from 'react-router';
+import Hardware from '../components/HardwareSetClass';
+import call, { Method } from '../utils/api';
+import HardwareDialog from '../components/HardwareDialog';
 
-interface ProjectViewProps {
-    token: Token;
+// Define an enum for error types
+enum ErrorType {
+    None = '',
+    PageOpenError = 'Failed to open project-view. Please try again.',
+    HardwareError ='No hardware set found.'
 }
 
-export default function ProjectView({ token }: Readonly<ProjectViewProps>) {
+
+const ProjectView = () => {
+    const {projectId} = useParams<{projectId: string}>();
+    const [projectName, setProjectName] = useState('');
+    const [description, setDescription] = useState('');
+    const [hardwareSet, setHardwareSets] = useState<Hardware[]>([]);
+    const [error, setError] = useState<ErrorType>(ErrorType.None); 
+        call(
+            `/api/projects/${projectId}`,
+            Method.Get,
+        )
+            .then(response => {
+                if (!response.ok) {
+                    setError(ErrorType.PageOpenError);
+                    return error;
+                }
+                return response.json();
+            })
+            .then((data: any) => {
+                setProjectName(data.name);
+                setDescription(data.description);
+                setHardwareSets(data.hardwaresets);  //I didnt see anything about this in the wiki but I am assuming this would return a list of hardwareId
```

Comment:
I don't like how the data is being stored here. You should just have a single `[project, setProject] = useState()` for the non hardware set stuff. 

```ts
inteface Project {
  id: string;
  name: string;
  description: string;
}
```